File: backend/src/main/java/se/matildaerenius/fluxo/mappers/TaskListMapper.java
package se.matildaerenius.fluxo.mappers;

import se.matildaerenius.fluxo.domain.dto.TaskListDto;
import se.matildaerenius.fluxo.domain.entities.TaskList;

public interface TaskListMapper {

    TaskList fromDto(TaskListDto taskListDto);
    TaskListDto toDto(TaskList taskList);
}

File: android-app/app/src/main/kotlin/se/matildaerenius/fluxo/data/remote/mapper/TaskListMapper.kt
package se.matildaerenius.fluxo.data.remote.mapper

import se.matildaerenius.fluxo.data.remote.dto.TaskListDto
import se.matildaerenius.fluxo.domain.model.TaskList

fun TaskListDto.toDomain(): TaskList {
    return TaskList(
        id = this.id ?: "",
        title = this.title,
        description = this.description
    )
}

fun TaskList.toDto(): TaskListDto {
    return TaskListDto(
        id = this.id.ifEmpty { null },
        title = this.title,
        description = this.description
    )
}